### PR TITLE
All Numba-dppy tests and examples using the @njit decorator should be used with "with_context"

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
Closes https://github.com/IntelPython/numba-dppy/issues/30

Changed one test for the example.

### CODE REVIEW

**Comment by reazulhoque** [comment]
@vlad-perevezentsev The changes are correct, let's change all the programs that need this change.

**Comment by PokhodenkoSA** [comment]
@vlad-perevezentsev Please, merge with main. It will fix CI.

**Comment by reazulhoque** [comment]
@vlad-perevezentsev Do you know what is causing the conflicts? Please let us know when the work is complete and we will start reviewing it.

**Review by PokhodenkoSA — APPROVED** [review]
LGTM. Will be merged after CI passes.

**Review by reazulhoque — APPROVED** [review]
LGTM

File: numba_dppy/tests/test_dpnp_functions.py [hunk 1]
```diff
@@ -75,132 +90,150 @@ def ensure_dpnp():
         return False
 
 
-@unittest.skipUnless(ensure_dpnp(), 'test only when dpNP is available')
-class Testdpnp_functions(DPPYTestCase):
+@unittest.skipUnless(ensure_dpnp() and dpctl.has_gpu_queues(), 'test only when dpNP and GPU is available')
+class Testdpnp_functions(DPPLTestCase):
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_dpnp_functions.py:94]:
```suggestion
class Testdpnp_functions(DPPYTestCase):
```

File: numba_dppy/tests/test_numpy_bit_twiddling_functions.py [hunk 2]
```diff
@@ -5,104 +5,115 @@
...
+import numba_dppy
+import numba_dppy as dppl
+import dpctl
 from numba_dppy.testing import unittest
 from numba_dppy.testing import DPPYTestCase
 
 
-class TestNumpy_bit_twiddling_functions(DPPYTestCase):
+@unittest.skipUnless(dpctl.has_gpu_queues(), 'test only on GPU system')
+class TestNumpy_bit_twiddling_functions(DPPLTestCase):
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_numpy_bit_twiddling_functions.py:16]:
```suggestion
class TestNumpy_bit_twiddling_functions(DPPYTestCase):
```

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Please, merge with main. Replace `DPPLTestCase` with `DPPYTestCase`. CI fails because of that.

File: numba_dppy/tests/test_vectorize.py [hunk 3]
```diff
@@ -5,21 +5,24 @@
 import sys
 import numpy as np
 from numba import njit, vectorize
-import numba_dppy, numba_dppy as dppy
+import numba_dppy
+import numba_dppy as dppl
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_vectorize.py:9]:
```suggestion
import numba_dppy as dppy
```

**Comment by PokhodenkoSA** [comment]
Windows fails possibly because Level0 is also available on CI machine and it is used by default.

**Comment by reazulhoque** [comment]
@vlad-perevezentsev  Please merge main to this PR and please mention the issues this PR is addresseing.

**Comment by PokhodenkoSA** [comment]
Applied comments from @diptorupd from https://github.com/IntelPython/numba-dppy/issues/65 in this commit https://github.com/IntelPython/numba-dppy/pull/46/commits/b34bb513f72b7be96b062c6b7615e9d92bf14c5d

**Review by PokhodenkoSA — APPROVED** [review]

**Pull request merged**